spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -88,11 +89,14 @@
 	 * @param parent the parent resource. For example:
 	 *     projects/{project_id}/databases/{database_id}/documents or
 	 *     projects/{project_id}/databases/{database_id}/documents/chatrooms/{chatroom_id}
+	 * @param mapper converter between Firestore document snapshots and Java objects
 	 */
-	public FirestoreTemplate(FirestoreStub firestore, String parent) {
+	public FirestoreTemplate(FirestoreStub firestore, String parent, FirestoreClassMapper mapper) {
 		this.firestore = firestore;
 		this.parent = parent;
 		this.databasePath = Util.extractDatabasePath(parent);
+		this.classMapper = mapper;
+		this.mappingContext = new FirestoreMappingContext(mapper);
```

Comment:
Calling `new` in the constructor is a code-smell because now you cannot configure your own version of the `mappingContext`.
The constructor should probably be changed to accept a `FirestoreMappingContext` bean instead of the mapper.